lib/src/index.rs

```diff
@@ -67,6 +73,17 @@ pub trait Index: Send + Sync {
 
     fn common_ancestors(&self, set1: &[CommitId], set2: &[CommitId]) -> Vec<CommitId>;
 
+    /// Heads among all indexed commits at the associated operation.
+    ///
+    /// Suppose the index contains all the historical heads and their
```

Comment:
This sounds clearer to me, thanks!

Optional nit: I'd split the sentence before "This returns..." (maybe "This function returns...").

I'd say it slightly differently:

"The index should contain all..." (or just "contains")


Possibly non-actionable thoughts: I guess one thing that wasn't clear to me is that, AFAICT, you consider pruning the index as part of pruning/reparenting of the operations tree. You could document that here. but I'm not sure this is the best place for it, and perhaps you already made this clear elsewhere.

Reply:
Made change: s/this/this function/

> "The index should contain all..." (or just "contains")

I avoided describing what the index *should* be because I don't know the details of cloud-based implementation. ;)

> I guess one thing that wasn't clear to me is that, AFAICT, you consider pruning the index as part of pruning/reparenting of the operations tree.

(Default)Index is basically a cache keyed by operation id, so yes, after reparenting, the new index won't include pruned commits. That's the reason why I decided to not insert tombstone into existing operation data.

I agree we'll need gc architecture doc somewhere.